app/controllers/MobileController.scala

```diff
@@ -0,0 +1,91 @@
+package controllers
+
+import java.sql.Timestamp
+import java.time.Instant
+import java.util.UUID
+import javax.inject.Inject
+
+import com.mohiva.play.silhouette.api.{Environment, Silhouette}
+import com.mohiva.play.silhouette.impl.authenticators.SessionAuthenticator
+import com.vividsolutions.jts.geom._
+import controllers.headers.ProvidesHeader
+import formats.json.CommentSubmissionFormats._
+import models.daos.slick.DBTableDefinitions.{DBUser, UserTable}
+import models.label.LabelTable
+import models.label.LabelTable.LabelValidationMetadata
+import models.mission.Mission
+import models.mission.MissionTable
+import models.validation._
+import models.user._
+import play.api.libs.json._
+import play.api.Logger
+import play.api.mvc._
+import models.amt.AMTAssignmentTable
+import models.label.LabelValidationTable
+
+
+import scala.concurrent.Future
+import scala.collection.mutable.ListBuffer
+
+class MobileController @Inject() (implicit val env: Environment[User, SessionAuthenticator])
+  extends Silhouette[User, SessionAuthenticator] with ProvidesHeader {
+  val gf: GeometryFactory = new GeometryFactory(new PrecisionModel(), 4326)
+
+  /**
+    * Returns the validation page.
+    * @return
+    */
+  def mobileValidate = UserAwareAction.async { implicit request =>
+    val timestamp: Timestamp = new Timestamp(Instant.now.toEpochMilli)
+    val ipAddress: String = request.remoteAddress
+
+    request.identity match {
+      case Some(user) =>
+        WebpageActivityTable.save(WebpageActivity(0, user.userId.toString, ipAddress, "Visit_MobileValidate", timestamp))
+        val possibleLabelTypeIds: ListBuffer[Int] = LabelTable.retrievePossibleLabelTypeIds(user.userId, 10, None)
+        val hasWork: Boolean = possibleLabelTypeIds.nonEmpty
+
+        // Checks if there are still labels in the database for the user to validate.
+        hasWork match {
```

Comment:
It looks like the `mobileValidate` function is basically identical to the `validate` function in `ValidationController.scala`. Is there a good reason not to have both of these call a function that does all the work and takes in an `isMobile` parameter?

Reply:
Check out what I did and make sure its ok